### PR TITLE
[wait for #1987] [TEST] Unit test case for NNFW ML-API

### PR DESCRIPTION
**Pull request by gichan-jang**
Unit tests are added to verify NNFW with ML-API.

Detail of added test case is :
 - Test multidimensional input/output us supported in NNFW
  - Single API, Invoke a sample model via Single API
  - Invoke a sample model via Pipeline API, with a stream input
  - Invoke a sample model via Pipeline API, with two input streams into a single tensor
 -  Invoke two sample model via Pipeline API, sharing a single input stream

NNFW has some problem with multidimensional input/output.
So this PR will be merged after fix the issue #9628(STAR/nnfw).

Signed-off-by: gichan-jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: []Passed [ ]Failed [* ]Skipped
2. Run test: []Passed [ ]Failed [* ]Skipped

### CODE REVIEW

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp [hunk 1]
```diff
@@ -11,6 +11,9 @@
 #include <glib.h>
 #include <glib/gstdio.h>        /* GStatBuf */
 #include <nnstreamer_plugin_api_filter.h>
+#include <nnstreamer.h>
+#include <nnstreamer-single.h>
+#include <nnstreamer-capi-private.h>
```
Comment (myungjoo) [inline tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: If you are going to test with Tizen ML APIs, ```nnstreamer-capi-private.h``` shouldn't be required.

Comment (gichan-jang) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: 'nnstreamer-capi-private.h' is included to use 'set_feature_state'.
'set_feature_state' need to construct the pipeline.

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp [hunk 2]
```diff
@@ -250,6 +253,369 @@ TEST (nnstreamer_nnfw_runtime_raw_functions, invoke)
...
+  test_model = get_model_file ();
+  ASSERT_TRUE (test_model != nullptr);
+
+  pipeline =
+      g_strdup_printf ("appsrc name=appsrc ! "
+      "other/tensor,dimension=(string)1:1:1:1,type=(string)float32,framerate=(fraction)0/1 ! "
+      "tensor_filter framework=nnfw model=%s ! "
+      "tensor_sink name=tensor_sink", test_model);
+
+  int status = ml_pipeline_construct_internal (pipeline, NULL, NULL, &handle);
```
Comment (jaeyun-jung) [inline tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: plz use ml_pipeline_construct(), not _internal()

Comment (gichan-jang) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: I updated what you mentioned.

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp [hunk 3]
```diff
@@ -236,20 +239,394 @@ TEST (nnstreamer_nnfw_runtime_raw_functions, invoke)
...
-  EXPECT_EQ (*((float *)output.data), 3.0);
+  EXPECT_EQ (*((float *) output.data), 3.0);
 
   sp->close (&prop, &data);
   g_free (model_file);
 }
 
+
+/**
+ * @brief Test nnfw subplugin with unsuccessful invoke (single ML-API)
```
Comment (dongju-chae) [inline tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: Is this really testing uncessessful invoke?

Comment (gichan-jang) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: Thank you for comment. It's my mistake.
It's successful invoke. I will modify it.

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp (hunk 1 above)
Comment (myungjoo) [reply tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: If that's all you use, then, it's perfectly fine :)

File: tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp [hunk 4]
```diff
@@ -236,20 +239,394 @@ TEST (nnstreamer_nnfw_runtime_raw_functions, invoke)
...
+    EXPECT_EQ (status, ML_ERROR_NONE);
+    g_usleep (100000);
+  }
+
+  ml_tensors_info_destroy (info);
+  ml_tensors_data_destroy (input);
+
+  status = ml_pipeline_stop (handle);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  g_usleep (50000);
```
Comment (kparichay) [inline tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp]: do we need this sleep?
Overall LGTM

**Comment by kparichay** [comment]
Can you also please apply the below diff to this file
```
--- a/tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp
+++ b/tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp
@@ -1,7 +1,7 @@
 /**
  * @file        unittest_tizen_capi.cpp
  * @date        13 Mar 2019
- * @brief       Unit test for Tizen CAPI of NNStreamer. Basis of TCT in the future.
+ * @brief       Unit test for Tizen nnfw tensor filter plugin.
  * @see         https://github.com/nnsuite/nnstreamer
  * @author      MyungJoo Ham <myungjoo.ham@samsung.com>
  * @bug         No known bugs
```

**Pull request closed without merging**